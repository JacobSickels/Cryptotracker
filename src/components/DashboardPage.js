import React from 'react';
import { connect } from 'react-redux';
import CryptoList from './CryptoList';
import { setDefaultFilter } from '../actions/filters';

/*
    DashboardPage is a React Component that is responsible for resetting the Redux filters
    to the current day when it is rendered, as well as rendering an instance of CryptoList.
*/

class DashboardPage extends React.Component {

    constructor(props) {
        super(props);
    }

    //Called before the component renders
    componentWillMount() {
        //Dashboard should only show the current day of data. 
        //This dispatch resets the filters to the current day
        this.props.setDefaultFilter();
    }

    render() {
        return (
            <div>
                <div className="page-container">
                    <CryptoList />
                </div>
            </div>
        );
    }
}

/*

    This function takes Redux dispatch actions and maps them to a function
    This new function is passed as a property in the component and is called in componentWillMount

    setDefaultFilter Redux action resets the startDate and endDate filters to the current day
*/
const mapDispatchToProps = (dispatch) => ({
    setDefaultFilter: () => dispatch(setDefaultFilter())
});

//This creates a Higher Order Component letting the component access the Redux dispatch actions
export default connect(null, mapDispatchToProps)(DashboardPage);