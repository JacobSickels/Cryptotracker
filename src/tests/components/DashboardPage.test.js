import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import DashboardPage from '../../components/DashboardPage';


test('should render dashboard page correctly', () => {
    const createMockStore = configureMockStore([thunk]);
    const store = createMockStore();
    const wrapper = shallow(
            <DashboardPage store={store}/>
        );
    expect(wrapper.dive()).toMatchSnapshot();
})