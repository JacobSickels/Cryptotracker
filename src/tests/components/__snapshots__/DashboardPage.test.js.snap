// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render dashboard page correctly 1`] = `
<div>
  <div
    className="page-container"
  >
    <CryptoList />
  </div>
</div>
`;
